frontend/js/src/playlists/PlaylistMenu.tsx

```diff
@@ -59,21 +64,27 @@ function PlaylistMenu({
       return;
     }
     try {
-      const newPlaylistId = await APIService.copyPlaylist(
-        currentUser.auth_token,
-        getPlaylistId(playlist)
-      );
+      let newPlaylistId;
+      if (playlistID) {
+        newPlaylistId = await APIService.copyPlaylist(auth_token, playlistID);
+      } else {
+        newPlaylistId = await APIService.createPlaylist(auth_token, {
+          playlist,
+        });
```

Comment:
I realized the LBRadio playlists do not have an identifier (they are temporary playlists that don't exist on the server) and trying to "Duplicate" was throwing an error, since `copyPlaylist` requires an existing playlist.

I implemented the mechanism we had in LBRadio.tsx previously, which uses `createPlaylist` instead to explicitly send a fully constructed playlist to be saved.